docker-compose.yml

```diff
@@ -1,12 +1,56 @@
 version: '2'
 services:
+  mongodb:
+    image: mongo
+    ports:
+      - "27017:27017"
+    volumes_from:
+      - mongodata
+    command: --smallfiles --rest --auth
+  mongodata:
+    image: mongo
+    volumes:
+      - /data/db
+    command: --break-mongo
+  redis:
+    image: redis
+    ports:
+      - "6379:6379"
+  elasticsearch:
+    image: elasticsearch
+    ports:
+      - "9200:9200"
+      - "9300:9300"
   db:
-    image: postgres
-  app:
+    image: mdillon/postgis
+    ports:
+      - "5432:5432"
+  railsweb:
     build: .
```

Comment:
I think you can use the Dockerfile to build this railsweb container, then reference it by name here. You would run something like `docker build -t railsweb .` to build the above container, then update this reference in docker-compose.yml to use the `railsweb` image. 

That should remove a lot of needed dependencies on the client machine itself, as the first container will contain them all.